packages/@aws-cdk/core/lib/custom-resource-provider/nodejs-entrypoint.ts

```diff
@@ -21,9 +21,27 @@ export type HandlerResponse = undefined | {
   NoEcho?: boolean;
 };
 
+/**
+ * Decodes encoded true/false values
+ */
+function decodeBooleans(object: object) {
```

Comment:
decode should only be called if `encodeValues` is enabled. I believe we will need to pass this information to the custom resource provider some how.

Reply:
i am supposed to move the `decodeBooleans` only to EKS, right? or do you want to do it only for the `custom-resources.Provider` and `custom-resources.AwsCustomResource`?

Reply:
Let's just keep this change only in the EKS module if that's possible.

Reply:
after starting to remove everything from the other libraries, i realized that by using the environment variables from the lambda themselves we can minimize the impact of the changes. see if we can still follow with the initial low-level approach like this, otherwise i will revert everything back

Reply:
Let’s revert please.